### PR TITLE
Add stop for JsonQueueWorker

### PR DESCRIPTION
**Pull request by janua**
When developing with the project `facia-end-to-end`, when compilation changes happen weird things happen with classes.

```
There is no started application
java.lang.RuntimeException: There is no started application
```

I reckon it has something to do with this class running without ever stopping.

@robertberry

### CODE REVIEW

**Comment by robertberry** [comment]
Ah yeah, good call. :+1:

File: facia-press/app/frontpress/JsonQueueWorker.scala [hunk 1]
```diff
@@ -128,4 +129,10 @@ abstract class JsonQueueWorker[A: Reads] extends Logging with ExecutionContexts
       }
     }
   }
+
+  final def stop(): Unit = {
+    synchronized {
```
Comment (gklopper) [inline facia-press/app/frontpress/JsonQueueWorker.scala]: :scream: - AtomicBoolean might be your friend here

**Comment by gklopper** [comment]
Have a look and see if you can rid yourself of the `synchronized` and `volatile` `var`s by swopping in `AtomicBoolean` and `AtomicReference`

**Comment by gklopper** [comment]
:+1:

**Comment by robertberry** [comment]
@gklopper The synchronized is to ensure the worker's only started once. Don't think you can do that with an AtomicBoolean ... 

There's no contention anyway, as it's only started once, when Play starts.

**Pull request merged**